File: targets/glfw/modules/native/monkeytarget.cpp

//***** monkeygame.h *****

class BBMonkeyGame : public BBGlfwGame{
public:

	static void Main( int args,const char *argv[] );
};

//***** monkeygame.cpp *****

#define _QUOTE(X) #X
#define _STRINGIZE(X) _QUOTE(X)

void Init_GL_Exts();

void BBMonkeyGame::Main( int argc,const char *argv[] ){

	if( !glfwInit() ){
		puts( "glfwInit failed" );
		exit(-1);
	}

	GLFWvidmode desktopMode;
	glfwGetDesktopMode( &desktopMode );
	
	int w=CFG_GLFW_WINDOW_WIDTH;
	if( !w ) w=desktopMode.Width;
	
	int h=CFG_GLFW_WINDOW_HEIGHT;
	if( !h ) h=desktopMode.Height;
	
	glfwOpenWindowHint( GLFW_WINDOW_NO_RESIZE,CFG_GLFW_WINDOW_RESIZABLE ? GL_FALSE : GL_TRUE );
	
	if( !glfwOpenWindow( w,h, 0,0,0,0,CFG_OPENGL_DEPTH_BUFFER_ENABLED ? 32 : 0,0,CFG_GLFW_WINDOW_FULLSCREEN ? GLFW_FULLSCREEN : GLFW_WINDOW  ) ){
		puts( "glfwOpenWindow failed" );
		exit(-1);
	}

	glfwSetWindowPos( (desktopMode.Width-w)/2,(desktopMode.Height-h)/2 );	

	glfwSetWindowTitle( _STRINGIZE(CFG_GLFW_WINDOW_TITLE) );

#if CFG_OPENGL_INIT_EXTENSIONS
	Init_GL_Exts();
#endif

	BBMonkeyGame *game=new BBMonkeyGame();
	
	try{
	
		bb_std_main( argc,argv );
		
	}catch( ThrowableObject *ex ){
	
		game->Die( ex );
		return;
	}

	if( game->Delegate() ) game->Run();
	
	glfwTerminate();
	
	exit( 0 );
}

File: targets/__deprecated__/xna/modules/native/monkeytarget.cs

class BBMonkeyGame : BBXnaGame{

	public BBMonkeyGame( Game app ) : base( app ){
	}
}

public class MonkeyGame : Game{

	BBMonkeyGame _game;

	public MonkeyGame(){
		_game=new BBMonkeyGame( this );
	}
	
	protected override void LoadContent(){
	
		try{
		
			bb_.bbInit();
			bb_.bbMain();
			
		}catch( Exception ex ){
		
			if( _game.Die( ex ) ) throw;
			return;
		}
		
		if( _game.Delegate()==null ){
			Exit();
			return;
		}
		
		_game.Run();
	}
	
	protected override void Update( GameTime gameTime ){
		_game.Update( gameTime );
		base.Update( gameTime );
	}
	
	protected override bool BeginDraw(){
		return _game.BeginDraw() && base.BeginDraw();
	}

	protected override void Draw( GameTime gameTime ){
		_game.Draw( gameTime );
		base.Draw( gameTime );
	}
	
#if !WINDOWS_PHONE
	public static void Main(){
		new MonkeyGame().Run();
	}
#endif
}
